contracts/GIVpower.sol

```diff
@@ -109,6 +113,21 @@ contract GIVpower is GardenUnipoolTokenDistributor, IERC20MetadataUpgradeable {
             revert CannotUnlockUntilRoundIsFinished();
         }
 
+        _unlock(accounts, round);
+    }
+
+    /// @notice Force unlock tokens belongs to accounts which are locked. The round does not need to be ended.
+    /// @dev Only owner can force unlock tokens
+    /// @param accounts List of accounts to unlock their tokens
+    /// @param round The round number token are locked till the end of
+    function forceUnlock(address[] calldata accounts, uint256 round) external onlyOwner {
+        _unlock(accounts, round);
+    }
+
+    /// @dev Internal function to unlock tokens and power for a specific account and round
+    /// @param accounts List of accounts to unlock their tokens
+    /// @param round The round number token are locked till the end of
+    function _unlock(address[] calldata accounts, uint256 round) internal {
```

Comment:
Do I understand that only the amounts that would unlock on the specified round are actually unlocked. 

In order to unlock all the locked tokens for a given user would you need to run this function multiple times specify a specific round or could you instead run this once by specifying a round that is one year from the current round?

if it is neither of those then how is it done? 

Do we need to know in which round a user's tokens would normally unlock?

Reply:
The first assumption, run once per round, is correct. In the subgraph database, we know which locked position the user has.

Reply:
@divine-comedian The efficient way is to aggregate accounts open positions over the round number. So, call the contract once per round, and on each call, pass all accounts with positive locked value released on that round.